docs/en/ingest-management/fleet/air-gapped.asciidoc

```diff
@@ -138,20 +146,66 @@ NODE_EXTRA_CA_CERTS="/etc/kibana/certs/ca-cert.pem"
-2. Place the tarball in a secure internal network location
-3. Use the <<elastic-agent-upgrade-command,`upgrade`>> command with a custom `--source-uri` option
-
-[discrete]
-=== Example
-
-["source", "sh", subs="attributes"]
+[[host-artifact-registry]]
+== Host your own artifact registry for binary downloads
+
+{agent}s must be able to access the {artifact-registry} to download
+binaries during upgrades. By default {agent}s download artifacts from
+`https://artifacts.elastic.co/downloads/`.
+
+To make binaries available in an air-gapped environment, you can host your own
+custom artifact registry, and then configure {agent}s to download binaries
+from it.
+
+. Create a custom artifact registry in a location accessible to your {agent}s:
+.. Download the latest release artifacts from the public {artifact-registry} at
+`https://artifacts.elastic.co/downloads/`. For example, the
+following cURL commands download all the artifacts that may be needed to upgrade
+{agent}s running on Linux. The exact list depends on which integrations you're
+using.
++
+["source","shell",subs="attributes"]
+----
+curl -O https://artifacts.elastic.co/downloads/apm-server/apm-server-{version}-linux-x86_64.tar.gz
+curl -O https://artifacts.elastic.co/downloads/beats/auditbeat/auditbeat-{version}-linux-x86_64.tar.gz
+curl -O https://artifacts.elastic.co/downloads/beats/elastic-agent/elastic-agent-{version}-linux-x86_64.tar.gz
+curl -O https://artifacts.elastic.co/downloads/beats/filebeat/filebeat-{version}-linux-x86_64.tar.gz
+curl -O https://artifacts.elastic.co/downloads/beats/heartbeat/heartbeat-{version}-linux-x86_64.tar.gz
+curl -O https://artifacts.elastic.co/downloads/beats/metricbeat/metricbeat-{version}-linux-x86_64.tar.gz
+curl -O https://artifacts.elastic.co/downloads/beats/osquerybeat/osquerybeat-{version}-linux-x86_64.tar.gz
+curl -O https://artifacts.elastic.co/downloads/beats/packetbeat/packetbeat-{version}-linux-x86_64.tar.gz
+curl -O https://artifacts.elastic.co/downloads/cloudbeat/cloudbeat-{version}-linux-x86_64.tar.gz
+curl -O https://artifacts.elastic.co/downloads/endpoint-dev/endpoint-security-{version}-linux-x86_64.tar.gz
+curl -O https://artifacts.elastic.co/downloads/fleet-server/fleet-server-{version}-linux-x86_64.tar.gz
+----
+.. On your HTTP file server, group the artifacts into directories and
+sub-directories that follow the same convention used by the {artifact-registry}:
++
+[source,shell]
 ----
-elastic-agent upgrade {version} --source-uri https://myinternaldomain.com/elastic-agent/elastic-agent-{version}-<platform>-x86_64.tar.gz
+<source_uri>/<artifact_type>/<artifact_name>-<version>-<arch>-<package_type>
 ----
++
+Where `<artifact_type>` may be `beats/elastic-agent`, `beats/filebeat`,
+`fleet-server`, `endpoint-dev`, and so on.
++
+[TIP]
+====
+Make sure you have a plan or automation in place to update your artifact
+registry when new versions of {agent} are available.
+====
+. Add the agent binary download location to {fleet} settings:
```

Comment:
Reviewers, a thread in a related PR reminded me that we need to account for standlone agents in these steps. Is the recommendation to set the source URI in the standalone config or during upgrade (when using the upgrade command)?

Reply:
@michalpristas Did you see this question. Should we try to cover standalone agents in this topic? If so, how do we want to recommend users configure the source URI?

Reply:
both are possible. 
setting in config will also affect downloading of other missing beats
setting in upgrade command will affect just upgrade procedure